File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Response/RevenueReportResponse.java
package ut.edu.be_quanlytro.Dto.Response;

import lombok.Builder;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Builder
public class RevenueReportResponse {
    private LocalDate period;               // Tháng thống kê (đã quy về ngày 1)
    private long totalInvoices;             // Tổng số lượng hóa đơn phát sinh trong tháng
    private long paidInvoicesCount;         // Số lượng hóa đơn ĐÃ THU
    private long unpaidInvoicesCount;       // Số lượng hóa đơn CÒN NỢ (Gồm UNPAID và OVERDUE)
    private BigDecimal totalCollectedAmount; // Tổng số tiền ĐÃ THU (Đã thanh toán)
    private BigDecimal totalDebtAmount;      // Tổng số tiền CÒN NỢ (Nợ xấu)
}

File: Flutter_QuanLyTro/lib/data/models/response/revenue_report_response.dart
class RevenueReportResponse {
  final DateTime? period;
  final int totalInvoices;
  final int paidInvoicesCount;
  final int pendingInvoicesCount;
  final int unpaidInvoicesCount;
  final double totalCollectedAmount;
  final double totalPendingAmount;
  final double totalDebtAmount;

  RevenueReportResponse({
    this.period,
    required this.totalInvoices,
    required this.paidInvoicesCount,
    required this.pendingInvoicesCount,
    required this.unpaidInvoicesCount,
    required this.totalCollectedAmount,
    required this.totalPendingAmount,
    required this.totalDebtAmount,
  });

  factory RevenueReportResponse.fromJson(Map<String, dynamic> json) {
    return RevenueReportResponse(
      period: json['period'] != null ? DateTime.parse(json['period']) : null,
      totalInvoices: json['totalInvoices'] ?? 0,
      paidInvoicesCount: json['paidInvoicesCount'] ?? 0,
      pendingInvoicesCount: json['pendingInvoicesCount'] ?? 0,
      unpaidInvoicesCount: json['unpaidInvoicesCount'] ?? 0,
      totalCollectedAmount: (json['totalCollectedAmount'] ?? 0).toDouble(),
      totalPendingAmount: (json['totalPendingAmount'] ?? 0).toDouble(),
      totalDebtAmount: (json['totalDebtAmount'] ?? 0).toDouble(),
    );
  }
}